### PR TITLE
Added AddRetryAfterErrorCondition() option

### PR DESCRIPTION
**Pull request by gtpaulose**
A really simple option so folks can set the retry options with easy and less verbosity in their code.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -554,6 +554,19 @@ func (c *Client) AddRetryCondition(condition RetryConditionFunc) *Client {
 	return c
 }
 
+// AddRetryAfterErrorCondition adds the basic condition of retrying after enncoutering
```
Comment (jeevatkm) [inline client.go:557]: @gtpaulose PR looks good, can you please correct typo `enncoutering` -> `encountering`?

File: client.go [hunk 2]
```diff
@@ -554,6 +554,19 @@ func (c *Client) AddRetryCondition(condition RetryConditionFunc) *Client {
...
+// an error from the http response
+func (c *Client) AddRetryAfterErrorCondition() *Client {
+	c.AddRetryCondition(func(response *Response, err error) bool {
+		if response.IsError() {
```
Comment (jeevatkm) [inline client.go:561]: @gtpaulose can you please add a test case for success and failure flow of new condition?
More info: https://codecov.io/gh/go-resty/resty/pull/384/diff?src=pr&el=tree#diff-Y2xpZW50Lmdv

**Review by jeevatkm — CHANGES_REQUESTED** [review]
@gtpaulose Thanks for the PR, surely this handy for the resty users.

Please take a look at a couple of comments.

File: client.go (hunk 2 above)
Comment (gtpaulose) [reply client.go:561]: Hey did you mean the test case for 'refiring under an error and non error scenario (i.e `response.IsError() == true` and `response.IsError() == false` )? '

Comment (moorereason) [reply client.go:561]: Yes

Comment (gtpaulose) [reply client.go:561]: Alright, ive added the test case. Ive reused an existing path in the test get server switch case, hope thats okay!

**Comment by gtpaulose** [comment]
Is there anymore changes required?

**Review by jeevatkm — APPROVED** [review]
@gtpaulose Thanks for the updates and for taking care of the review comments. 
I'm really sorry for the delay, due to personal reasons.

**Pull request merged**